src/group_theory/group_action/prod.lean

```diff
@@ -45,6 +46,17 @@ instance has_faithful_scalar_right [nonempty α] [has_faithful_scalar M β] :
 
 end
 
+@[to_additive]
+instance smul_comm_class_both [monoid N] [monoid P] [has_scalar M N] [has_scalar M P]
+  [smul_comm_class M N N] [smul_comm_class M P P] :
+  smul_comm_class M (N × P) (N × P) :=
+⟨λ c x y, by simp [smul_def, mul_def, mul_smul_comm]⟩
```

Comment:
Strictly we should also have the
```suggestion
@[to_additive]
instance smul_comm_class_both' [monoid N] [monoid P] [has_scalar M N] [has_scalar M P]
  [smul_comm_class N M N] [smul_comm_class P M P] :
  smul_comm_class (N × P) M (N × P) :=
sorry
```
version

Reply:
I prefer to add instances as we need them.